practice/Capstone/main.py

```diff
@@ -0,0 +1,249 @@
+import argparse
+import configparser
+import json
+import logging
+import os
+import random
+import re
+import time
+import uuid
+from uuid import uuid4
+import sys
+import multiprocessing
+from functools import partial
+from concurrent.futures import ThreadPoolExecutor
+from typing import List, Dict, Any, Callable, NamedTuple
+
+
+class Args(NamedTuple):
+    path_to_save_files: str
+    files_count: int
+    file_prefix: str
+    file_name: str
+    data_lines: int
+    multiprocessing: int
+    data_schema: str
+    clear_path: bool
+
+    def check_values(self):
+        if self.files_count > 1 and not self.file_prefix:
+            raise ValueError('If files_count > 1, then file_prefix argument needs to be provided.')
+        if self.data_lines < 1:
+            raise ValueError("Number of data lines must be at least 1")
+        if self.multiprocessing < 1:
+            raise ValueError("Number of processes must be at least 1")
+        if self.files_count < 0:
+            raise ValueError('Number of files cannot be negative.')
+        return
+
+
+def return_timestamp_fun(value: str) -> Callable:
```

Comment:
usually people use 'func' for short function namings. just a note, because a bit unclear)

Reply:
Done.